examples/direct_fidelity_estimation.py

```diff
@@ -260,11 +261,11 @@ def direct_fidelity_estimation(circuit: cirq.Circuit, qubits: List[cirq.Qid],
     # above) does bother it, so we re-normalize the probs.
     p /= np.sum(p)
 
-    noisy_simulator = cirq.DensityMatrixSimulator(noise=noise)
     fidelity = 0.0
 
     if samples_per_term == 0:
         # sigma in https://arxiv.org/abs/1104.3835
+        noisy_simulator = cast(cirq.DensityMatrixSimulator, sampler)
```

Comment:
We should raise an exception if sampler is not of the right type.

Reply:
Added an assert

Reply:
Asserts should be used for conditions that are true regardless of input parameters due to internal consistency of the algorithm (e.g. [loop invariants](https://en.wikipedia.org/wiki/Loop_invariant)). Assert failure should indicate a bug.

In this case, the condition may be false due to user input. This should be handled by raising an exception that tells the caller what they did wrong.

Reply:
This isn't done, see comment two lines above.

Reply:
Sorry. I should have been more careful. How about now?